### PR TITLE
testcne: add functional tests for cne lib to testcne

### PR DESCRIPTION
**Pull request by sushmasi**
Add functional tests relevant to lib/core/cne/cne.[ch]
to testcne. Functions which are covered by other testcne
tests are not covered.

Signed-off-by: Sushma Sitaram <sushma.sitaram@intel.com>

### CODE REVIEW

File: test/testcne/cne_setup_test.c [hunk 1]
```diff
@@ -0,0 +1,123 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright (c) 2019-2022 Intel Corporation
```
Comment (KeithWiles) [inline test/testcne/cne_setup_test.c:2]: Think this should be only 2022

Comment (KeithWiles) [reply test/testcne/cne_setup_test.c:2]: maybe the file name should be 'cne_register_test.c'

File: test/testcne/cne_setup_test.c [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+static int
+cne_test(void)
+{
+    int tuid, cuid, tcnt, ret = 0;
+    int v = 1234, vv = 0;
+    void *setv  = &v;
+    void *tempv = &vv;
+    void **getv = &tempv;
+
+    tuid = cne_register("TestCNE");
```
Comment (KeithWiles) [inline test/testcne/cne_setup_test.c:31]: TestCNE --> TestCNERegister

File: test/testcne/cne_setup_test.c [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    return 0;
+
+err:
+    if (tuid)
+        cne_unregister(tuid);
+    return -1;
+}
+
+int
+cne_main(int argc, char **argv)
```
Comment (KeithWiles) [inline test/testcne/cne_setup_test.c:93]: Maybe 'cne_register_main' is a better name.

File: test/testcne/cne_setup_test.c [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        case 'V':
+            verbose = 1;
+            break;
+        default:
+            break;
+        }
+    }
+    CNE_SET_USED(verbose);
+
+    tst = tst_start("CNE Setup");
```
Comment (KeithWiles) [inline test/testcne/cne_setup_test.c:113]: CNE Setup --> CNE Registration

File: test/testcne/cne_setup_test.c [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+#include "cne_setup_test.h"
+
+static void
+__on_exit(int sig __cne_unused, void *arg __cne_unused, int exit_type __cne_unused)
+{
+    return;
+}
+
+static int
+cne_test(void)
```
Comment (KeithWiles) [inline test/testcne/cne_setup_test.c:23]: cne_register_test is a better name.

File: test/testcne/cne_setup_test.h [hunk 6]
```diff
@@ -0,0 +1,24 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright (c) 2019-2022 Intel Corporation
+ */
+
+#ifndef _CNE_SETUP_TEST_H_
```
Comment (KeithWiles) [inline test/testcne/cne_setup_test.h:5]: CNE_REGISTER_TEST_H

File: test/testcne/cne_setup_test.h [hunk 7]
```diff
@@ -0,0 +1,24 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright (c) 2019-2022 Intel Corporation
```
Comment (KeithWiles) [inline test/testcne/cne_setup_test.h:2]: Copyright 2022 only

Comment (KeithWiles) [reply test/testcne/cne_setup_test.h:2]: Filename should be 'cne_register_test.h'

File: test/testcne/cne_setup_test.h [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+ * @file
+ * CNE Setup Test
+ *
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+int cne_main(int argc, char **argv);
```
Comment (KeithWiles) [inline test/testcne/cne_setup_test.h:18]: cne_main --> cne_register_main

File: test/testcne/cne_setup_test.h [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif /* _CNE_SETUP_TEST_H_ */
```
Comment (KeithWiles) [inline test/testcne/cne_setup_test.h:24]: Update comment

File: test/testcne/meson.build [hunk 10]
```diff
@@ -16,6 +16,7 @@ subdir('files/json') # Copy files for jcfg test
 # Keep lists sorted
 sources = files(
     'acl_test.c',
+    'cne_setup_test.c',
```
Comment (KeithWiles) [inline test/testcne/meson.build:19]: rename cne_register_test.c

**Review by KeithWiles — CHANGES_REQUESTED** [review]
Looks pretty good, I think the test name needs to change.

**Comment by sushmasi** [comment]
> Looks pretty good, I think the test name needs to change.

Any suggestions for the test name? Going with the pattern of other tests under testcne, cne_test.[ch] were giving build errors due to the reference to an existing cne_test.h under lib/include

**Comment by KeithWiles** [comment]
reverify

**Comment by jeffreybshaw** [comment]
@sushmasi can you also run "clang-format" before re-submitting this patch? When you added a new include to cli_cmds.h<sup>1</sup>, clang-format wants to adjust the spaces before the comments.

Obviously this is not very important, but I'd prefer to have clang-format run to be clean.

<sup>1</sup> https://github.com/CloudNativeDataPlane/cndp/runs/6558534623?check_suite_focus=true

**Review by KeithWiles — APPROVED** [review]
LGTM

**Pull request merged**